File: Backend/bank/src/main/java/com/Bank/service/UserService.java
package com.Bank.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.Bank.dto.UserDTO;
import com.Bank.repository.UserRepositoryIn;


@Service
public class UserService {
	private final UserRepositoryIn userRepository;
	

    @Autowired
    public UserService(UserRepositoryIn  userRepository) {
        this.userRepository = userRepository;
		
    }
	   // method 
	public UserDTO createUser(UserDTO user) {
		return userRepository.save(user);

	}

	 public UserDTO getUserByUsername(String username) {
		 
		 	
	        return userRepository.findByUsername(username);
	    
	 }
		
	
}

//services uses @Service annotion 
//we call repositry inside services 
//after this we create controller 

File: Frontend/Bank_ui/src/Services/UserService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/user/add';

const UserService = {
    getAllUsers: () => {
        return axios.get(API_BASE_URL);
    },

    getUserByUsername: (username) => {
        return axios.get(API_BASE_URL + username);
    }

};

export default UserService;
